pkg/cluster/config.go

```diff
@@ -0,0 +1,27 @@
+package cluster
+
+import (
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+type Config struct {
+	Cluster               *anywherev1.Cluster
+	VSphereDatacenter     *anywherev1.VSphereDatacenterConfig
+	DockerDatacenter      *anywherev1.DockerDatacenterConfig
+	VSphereMachineConfigs map[string]*anywherev1.VSphereMachineConfig
+	OIDCConfigs           map[string]*anywherev1.OIDCConfig
+	AWSIAMConfigs         map[string]*anywherev1.AWSIamConfig
+	GitOpsConfig          *anywherev1.GitOpsConfig
+}
+
+func (c *Config) VsphereMachineConfig(name string) *anywherev1.VSphereMachineConfig {
```

Comment:
I'm still on the fence about this.

I like the idea of hiding the fact that we store this as a map. It allow us that change that later and it also stops other packages from adding more elements.

But if we want to give the ability to iterate over the whole collection, we would need another method that returns it, which adds "extra work" when adding new providers.

Opinions?